consoles/s3270.pm

```diff
@@ -144,100 +230,11 @@ sub expect_3270 ($self, %arg) {
-
-            # if there is MORE..., go and grab it.
-            if ($status_line =~ /$arg{buffer_full}/) {
-                $self->ensure_screen_update();
-                next;
-            }
-
-            if ($status_line !~ /$arg{expected_status}/) {
-                # if the timeout is not over, wait for more output
-                my $elapsed_time = time() - $start_time;
-                next if $elapsed_time < $arg{timeout} && $self->wait_output($arg{timeout} - $elapsed_time);
-
-                # flush the buffer for debugging:
-                while (my $line = $self->{raw_expect_queue}->dequeue_nb()) {
-                    push @$result, $line;
-                }
-                confess "expect_3270: timed out waiting for 'expected_status'.\n"
-                  . "  waiting for ${\Dumper \%arg}\n"
-                  . "  last output:\n"
-                  . Dumper($result)
-                  . $status_line;
-            }
-
-            die "status line must match 'buffer_ready'" unless ($status_line =~ /$arg{buffer_ready}/);
-        }
-
-        # No more host output is pending. We have some output in the raw_expect_queue,
-        # possibly from a previous run
-
-        # If *not* looking for output_delim, we reached 'buffer_ready' and just return the output buffer
-        if (!defined $arg{output_delim}) {
-            # no need to wait for something special. We just return what we have...
-            while (my $line = $self->{raw_expect_queue}->dequeue_nb()) {
-                push @$result, $line;
-            }
-            last;
-        }
-
-        my $line;
-        while ($line = $self->{raw_expect_queue}->dequeue_nb()) {
-            push @$result, $line;
-            last if !defined $line || $line =~ /$arg{output_delim}/;
-        }
-
-        # If we matched the 'output_delim', we are done.
-        last if defined $line;
-
-        # The queue is empty. If we got so far and we had some output on the
-        # screen the last time, clear the screen so we don't grab the same
-        # stuff again.
-
-        # TODO The better alternative solution to the same problem
-        # would be to remember lines that were not updated since the
-        # last Snap(Ascii) and to thus avoid duplicate lines.
-
-        # For now we have to live with having a clear screen.
-        $self->ensure_screen_update() if $we_had_new_output;
-
-        # wait for new output from the host.
-        my $elapsed_time = time() - $start_time;
-        if ($elapsed_time > $arg{timeout}
-            || !$self->wait_output($arg{timeout} - $elapsed_time))
-        {
-            confess "expect_3270: timed out.\n" . "  waiting for ${\Dumper \%arg}\n" . "  last output:\n" . Dumper($result);
-        }
-    }
-
+    my $do_run = 1;
     # tracing output
+    while ($do_run) {
+        $do_run = $self->_handle_expect_3270_cycle($result, time(), %arg);
+    }
```

Comment:
Without the loop body the this comment is now lacking context so I'd just remove it. You can now also avoid the variable and loop body.

```suggestion
    while ($self->_handle_expect_3270_cycle($result, time(), %arg));
```

Reply:
Well, my Perl says `while (...);` is not valid syntax but at least we can shorten it to `while (...) {};`

Reply:
That is a disappointment :-)
We can go for `{}` then.